server/src/sas/CompletionProvider.ts

```diff
@@ -281,6 +284,381 @@ export class CompletionProvider {
+                      if (_arguments.length !== 1) {
+                        _arguments = _arguments.filter((item) => item);
+                      }
+                      _arguments = [...new Set(_arguments)];
+
+                      const argumentsInData = data.arguments?.map(
+                        (item) =>
+                          item.name
+                            ?.replace("(s)", "")
+                            .replace(/['"‘’<>()|.]/g, "")
+                            .replace(/[-_\s]?\d$/, "")
+                            .replace(/[-_\s]n$/, "")
+                            .trim()
+                            .toLowerCase(),
+                      );
+                      const descriptions = _arguments
+                        .map((_argument) => {
+                          let argumentInData = data.arguments?.find(
+                            (_item, _index) => {
+                              if (_argument === argumentsInData?.[_index]) {
+                                allArguments.push(_item.name);
+                                return true;
+                              }
+                            },
+                          );
+                          if (!argumentInData) {
+                            argumentInData = data.arguments?.find(
+                              (_item, _index) => {
+                                if (
+                                  _argument.split(" ")?.[0] ===
+                                    argumentsInData?.[_index].split(" ")?.[0] ||
+                                  _argument.split("_")?.[0] ===
+                                    argumentsInData?.[_index].split("_")?.[0] ||
+                                  _argument.split("-")?.[0] ===
+                                    argumentsInData?.[_index].split("-")?.[0] ||
+                                  _argument.split("–")?.[0] ===
+                                    argumentsInData?.[_index].split("–")?.[0]
+                                ) {
+                                  allArguments.push(_item.name);
+                                  return true;
+                                }
+                              },
+                            );
+                          }
+                          if (argumentInData) {
+                            return `**${argumentInData.name}:** ${argumentInData.description}`;
+                          }
+                          return "";
+                        })
+                        .filter((item) => item);
+
+                      if (_arguments.length === 1) {
+                        return [
+                          argument.includes("(s)")
+                            ? argument.replace(/[<>]/g, "").trim()
+                            : argument.replace(/[<>()]/g, "").trim(),
+                          descriptions[0] ?? "",
+                          true,
+                        ] as [string, string, boolean];
```

Comment:
avoid using `as` as long as possible.
declare the type as `.map<[string, string, boolean]>()`